### PR TITLE
Converting remaining anonymous Callable and FileCallable implementations to named classes

### PR DESCRIPTION
**Pull request by jglick**
Trying to take care of a bunch of log warnings in bulk.

### CODE REVIEW

File: core/src/main/java/hudson/FilePath.java [hunk 1]
```diff
@@ -748,17 +793,25 @@ public OutputStream compress(OutputStream out) throws IOException {
...
-            });
+            act(new UntarFrom(compression, in));
         } finally {
             _in.close();
         }
     }
+    private class UntarFrom extends SecureFileCallable<Void> {
+        private final TarCompression compression;
+        private final InputStream in;
+        UntarFrom(TarCompression compression, InputStream in) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: Must be `RemoteInputStream` IMHO

File: core/src/main/java/hudson/FilePath.java [hunk 2]
```diff
@@ -661,14 +696,22 @@ public String invoke(File f, VirtualChannel channel) throws IOException {
...
-                Util.createSymlink(f.getParentFile(),target,f.getName(),listener);
-                return null;
-            }
-        });
+        act(new SymlinkTo(target, listener));
+    }
+    private class SymlinkTo extends SecureFileCallable<Void> {
+        private final String target;
+        private final TaskListener listener;
+        SymlinkTo(String target, TaskListener listener) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: Is TaskListener Remotable OOTB?

File: core/src/main/java/hudson/FilePath.java [hunk 3]
```diff
@@ -578,13 +606,19 @@ public Void invoke(File dir, VirtualChannel channel) throws IOException, Interru
...
-                unzip(dir, in);
-                return null;
-            }
-            private static final long serialVersionUID = 1L;
-        });
+        act(new UnzipFrom(in));
+    }
+    private class UnzipFrom extends SecureFileCallable<Void> {
+        private final InputStream in;
+        UnzipFrom(InputStream in) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: RemoteInputStream

File: core/src/main/java/hudson/FilePath.java [hunk 4]
```diff
@@ -478,7 +478,18 @@ public int zip(OutputStream out, DirScanner scanner) throws IOException, Interru
...
     public int archive(final ArchiverFactory factory, OutputStream os, final DirScanner scanner) throws IOException, InterruptedException {
         final OutputStream out = (channel!=null)?new RemoteOutputStream(os):os;
-        return act(new SecureFileCallable<Integer>() {
+        return act(new Archive(factory, out, scanner));
+    }
+    private class Archive extends SecureFileCallable<Integer> {
+        private final ArchiverFactory factory;
+        private final OutputStream out;
+        private final DirScanner scanner;
+        Archive(ArchiverFactory factory, OutputStream out, DirScanner scanner) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: RemoteOutputStream

File: core/src/main/java/hudson/FilePath.java [hunk 5]
```diff
@@ -478,7 +478,18 @@ public int zip(OutputStream out, DirScanner scanner) throws IOException, Interru
      */
     public int archive(final ArchiverFactory factory, OutputStream os, final DirScanner scanner) throws IOException, InterruptedException {
         final OutputStream out = (channel!=null)?new RemoteOutputStream(os):os;
-        return act(new SecureFileCallable<Integer>() {
+        return act(new Archive(factory, out, scanner));
+    }
+    private class Archive extends SecureFileCallable<Integer> {
+        private final ArchiverFactory factory;
+        private final OutputStream out;
+        private final DirScanner scanner;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: It's a bad idea to serialize it over the channel IMHO. Add better APIs and deprecate the existing method?

File: core/src/main/java/hudson/FilePath.java [hunk 6]
```diff
@@ -2079,13 +2192,19 @@ public void renameTo(final FilePath target) throws IOException, InterruptedExcep
...
-            private static final long serialVersionUID = 1L;
-            public Void invoke(File f, VirtualChannel channel) throws IOException {
-                Files.move(fileToPath(reading(f)), fileToPath(creating(new File(target.remote))), LinkOption.NOFOLLOW_LINKS);
-                return null;
-            }
-        });
+        act(new RenameTo(target));
+    }
+    private class RenameTo extends SecureFileCallable<Void> {
+        private final FilePath target;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: Seems like a bad conversion. I would rather to do the `target.remore`conversion outside and use `File` as argument

File: core/src/main/java/hudson/FilePath.java [hunk 7]
```diff
@@ -2097,8 +2216,15 @@ public void moveAllChildrenTo(final FilePath target) throws IOException, Interru
         if(this.channel != target.channel) {
             throw new IOException("pullUpTo target must be on the same host");
         }
-        act(new SecureFileCallable<Void>() {
+        act(new MoveAllChildrenTo(target));
+    }
+    private class MoveAllChildrenTo extends SecureFileCallable<Void> {
+        private final FilePath target;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: same as above?

File: core/src/main/java/hudson/FilePath.java [hunk 8]
```diff
@@ -2157,6 +2273,21 @@ public Void invoke(File f, VirtualChannel channel) throws IOException {
         target.chmod(mode());
         target.setLastModifiedIfPossible(lastModified());
     }
+    private class CopyToWithPermission extends SecureFileCallable<Void> {
+        private final FilePath target;
+        CopyToWithPermission(FilePath target) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: same as above

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Was it an automatic refactoring? 

IMHO some static classes could use stricter APIs. `FilePath` is likely safe (do not recall), but it would be better to just resolve methods to `FilePath#remote` where possible.

Will do a deeper review next week

**Comment by jglick** [comment]
> Was it an automatic refactoring?

No, IDE-assisted on each class.

> some static classes could use stricter APIs. `FilePath` is likely safe

It is.

> it would be better to just resolve methods to `FilePath#remote` where possible

Of course that would be nice, if it were straightforward. Unfortunately, it is not, due to the file path filter handling. https://github.com/jenkinsci/jenkins/pull/3333#pullrequestreview-102076778 gives a full explanation.

File: core/src/main/java/hudson/FilePath.java (hunk 5 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: The `DirScanner` needs to be serialized by the nature of this API.

File: core/src/main/java/hudson/FilePath.java (hunk 4 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: Not if `channel == null`.

File: core/src/main/java/hudson/FilePath.java (hunk 3 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: Does not really matter.

File: core/src/main/java/hudson/FilePath.java (hunk 2 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: Yes it is.

File: core/src/main/java/hudson/FilePath.java (hunk 1 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: This is a `private` class. The caller passes a working stream.

File: core/src/main/java/hudson/FilePath.java (hunk 6 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: For purposes of this PR I am not touching @kohsuke’s logic. Whether `creating(new File(target.remote))` should be replaced with `target.creating(new File(target.remote))` I am not sure. If you care to propose such changes, no problem, but they should not buried in a simple extract-nested-class refactoring like this.

File: core/src/main/java/hudson/FilePath.java (hunk 7 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: ditto

File: core/src/main/java/hudson/FilePath.java (hunk 8 above)
Comment (jglick) [reply core/src/main/java/hudson/FilePath.java]: ditto

**Review by oleg-nenashev — APPROVED** [review]
All comments are about private classes, so it does not matter much. Feel free to ship it

**Comment by oleg-nenashev** [comment]
@jglick could you please merge with master to retrigger CI?

**Pull request merged**